uddl_test/uddl_test_formatting/universal-data-definition-language-formatter.test.ts

```diff
@@ -2,24 +2,217 @@ import { createUniversalDataDefinitionLanguageServices } from "../../src/languag
+      after: `dm PPT "Base data structures to support People, Places and Things"
+{
+
+    cdm Conceptual "Need to start at Conceptual Level"
+    {
+    
+        basis uddlBasis "Formating conceptual basis entity";
+    }
+    
+}`,
+    });
+  });
+
+  it("should format cdm with conceptual domain", async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before:
+        'dm PPT "Base data structures to support People, Places and Things"{cdm Conceptual "Need to start at Conceptual Level"{domain NaturalPerson "Base definition of a natural person"; }}',
+      after: `dm PPT "Base data structures to support People, Places and Things"
+{
+
+    cdm Conceptual "Need to start at Conceptual Level"
+    {
+    
+        domain NaturalPerson "Base definition of a natural person";
+    }
+    
+}`,
+    });
+  });
+
+  it("should format ldm array", async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before:
+        'dm Book "Base definition for book requirement" {ldm library "contains available books" {ldm BookShelf "Books related by category"{}}}',
+      after: `dm Book "Base definition for book requirement"
+{
+
+    ldm library "contains available books"
+    {
+    
+        ldm BookShelf "Books related by category"
+        {
+        
+        }
+        
+    }
+    
+}`,
+    });
+  });
+
+  it("should format logical element", async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before:
+        'dm logicalElement "Some logial element" {ldm logicalData "logical description"{lunit LengthUnit "Defines units of length";bool IsEnabled "Represents whether an item is enabled";nat NumberOfChildren "Represents the number of children a person has";}}',
+      after: `dm logicalElement "Some logial element"
+{
+
+    ldm logicalData "logical description"
+    {
+    
+        lunit LengthUnit "Defines units of length";
+        bool IsEnabled "Represents whether an item is enabled";
+        nat NumberOfChildren "Represents the number of children a person has";
+    }
+    
+}`,
+    });
+  });
+
+  it("should format logical relationships",async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before:'dm logicalElement "Some logial element" {ldm logicalData "logical description"{csa logicalCoordinate "Logical system axis";coord GeographicCoordinates "Represents geographic coordinates" {axis:"WGS84" angleEq:"lat = atan((exp(n) - exp(-n)) / (exp(n) + exp(-n)))" distanceEq:"distance = R * arctan(sqrt(1 - e^2) * tan(lat))"[logicalCoordinate]};}}',
+      after:`dm logicalElement "Some logial element"
+{
+
+    ldm logicalData "logical description"
+    {
+    
+        csa logicalCoordinate "Logical system axis";
+        coord GeographicCoordinates "Represents geographic coordinates" {
+        
+            axis:
```

Comment:
I suggest that the prompt and its value be on the same line ( `axis: "WGS84"`). That will be easier to understand.